tests/e2e/tests/conftest.py

```diff
@@ -32,3 +34,50 @@ def h() -> harness.Harness:
 
     LOG.debug("Cleanup")
     h.cleanup()
+
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers",
+        "node_count: Mark a test to specify how many instance nodes need to be created",
+    )
+
+
+@pytest.fixture(scope="function")
+def node_count(request) -> int:
+    node_count_marker = request.node.get_closest_marker("node_count")
+    if not node_count_marker:
+        return 1
+    node_count_arg, *_ = node_count_marker.args
+    return int(node_count_arg)
+
+
+@pytest.fixture(scope="function")
+def instances(
+    h: harness.Harness, node_count: int, tmp_path: Path
+) -> List[harness.Instance]:
+    """Construct instances for a cluster.
+
+    Bootstrap and setup networking on the first instance.
+    """
+    if not config.SNAP:
+        pytest.fail("Set TEST_SNAP to the path where the snap is")
+
+    if not node_count:
+        pytest.xfail("Test requested 0 instances, skip this test.")
+
+    snap_path = (tmp_path / "k8s.snap").as_posix()
+
+    LOG.info(f"Creating {node_count} instances")
+    instances: List[util.Instance] = []
+
+    for _ in range(node_count):
+        # Create <node_count> instances and setup the k8s snap in each.
+        instance = h.new_instance()
+        instances.append(instance)
+        util.setup_k8s_snap(instance, snap_path)
+
+    instances[0].exec(["k8s", "bootstrap"])
+    util.setup_network(instances[0])
```

Comment:
nitpick: node_count() uses unpacking syntax to access a single element. Do you want to do the same thing here ?
i.e. instance, *_ = instances